libpod/kube.go

```diff
@@ -233,6 +233,61 @@ func GenerateForKubeDeployment(ctx context.Context, pod *YAMLPod, options entiti
 	return &dep, nil
 }
 
+// GenerateForKubeJob returns a YAMLDeployment from a YAMLPod that is then used to create a kubernetes Job
+// kind YAML.
+func GenerateForKubeJob(ctx context.Context, pod *YAMLPod, options entities.GenerateKubeOptions) (*YAMLJob, error) {
+	// Restart policy for Job cannot be set to Always
+	if options.Type == define.K8sKindJob && pod.Spec.RestartPolicy == v1.RestartPolicyAlways {
+		return nil, fmt.Errorf("k8s Jobs can not have restartPolicy set to Always; only Never and OnFailure policies allowed")
+	}
+
+	// Create label map that will be added to podSpec and Job metadata
+	// The matching label lets the job know which pods to manage
+	appKey := "app"
+	matchLabels := map[string]string{appKey: pod.Name}
+	// Add the key:value (app:pod-name) to the podSpec labels
+	if pod.Labels == nil {
+		pod.Labels = matchLabels
+	} else {
+		pod.Labels[appKey] = pod.Name
+	}
+
+	jobSpec := YAMLJobSpec{
+		Template: &YAMLPodTemplateSpec{
+			PodTemplateSpec: v1.PodTemplateSpec{
+				ObjectMeta: pod.ObjectMeta,
+			},
+			Spec: pod.Spec,
+		},
+	}
+
+	// Set the completions and parallelism to 1 by default for the Job
+	completions, parallelism := int32(1), int32(1)
+	jobSpec.Completions = &completions
+	jobSpec.Parallelism = &parallelism
```

Comment:
Why does generate add these values as one but then kube play always throws a warning when they are set?
logrus.Warnf("Ignoring completions and/or parallelism values as it is not supported by Podman") 

Does that not mean a yaml generated by podman will always throw an error when played again or I am missreading the code here?
Either the warning should not be printed when the value is 1 or generate should not add it.

Reply:
We should not warn when value is 1.

Reply:
We throw a warning because the values for parallelism and completions have no meaning in podman world - we do something similar with replicas for deployments. I am fine with removing the warning and adding a note to the docs that these are ignored.

Reply:
Fixed.

Reply:
That warning should allow 1 as valid value not just do  != nil or you cannot set these values on generate.

Reply:
thanks missed it in the other location, fixed now.